src/test/steps/loginSteps.ts

```diff
@@ -1,37 +1,42 @@
 import {Given, When, Then} from '@cucumber/cucumber';
-import {chromium, Page, Browser, expect} from '@playwright/test';
-
-let browser: Browser;
-let page: Page;
-
-Given('User navigates to the application', async function () {
-  browser = await chromium.launch({headless: false});
-  page = await browser.newPage();
-  page.goto('https://px-stage.self-service.danads.com');
+import {pageFixture} from '../../hooks/pageFixture';
+import Assert from '../../helper/wrapper/assert';
+import AuthPage from '../../pages/basicAuthPage';
+import LoginPage from '../../pages/loginPage';
+
+let authPage: AuthPage;
+let loginPage: LoginPage;
+let assert: Assert;
+
+Given('User passes the authorization', async function () {
+  authPage = new AuthPage(pageFixture.page);
+  assert = new Assert(pageFixture.page);
+  await authPage.authorizeUser();
+  await assert.assertTitleContains('Core Ad Manager');
 });
 
-Given('User clicks login button', async function () {
-  //await page.locator("//a/*[text()='Log in']").click();
+Given('User clicks Log in button', async function () {
+  loginPage = new LoginPage(pageFixture.page);
+  await loginPage.clickLogInButton();
 });
 
-Given('User enters username', async function () {
-  await page.locator('(//input[@name=\'username\'])[2]').fill('Username');
+When('User enters email', async function () {
+  await loginPage.enterEmail(process.env.ADMIN_EMAIL);
 });
```

Comment:
What if you need to authenticate as publisher, or advertiser, or agency?

Reply:
Now we don't need other user roles or we can get their credentials from the .env file

Reply:
Fixed